text/0000-asset-manifest.md

```diff
@@ -0,0 +1,181 @@
+- Start Date: 2016-07-17
+- RFC PR: [emberjs/rfcs#153](https://github.com/emberjs/rfcs/pull/153)
+- Ember Issue: (leave this empty)
+
+# Summary
+
+The Asset Manifest is a JSON specification to describe assets and bundles of assets that can be loaded into an Ember application asynchronously at runtime.
+
+This is a corollary to a forthcoming Asset Loader service which will handle the asynchronous loading of application assets.
+
+# Motivation
+
+The primary motivation for defining an Ember Asset Manifest specification is to enable [lazy-loading of Engines](https://github.com/emberjs/rfcs/blob/master/text/0010-engines.md#lazy-loading-manifests). Beyond that it can also be used for any piece-wise loading strategy of assets into an Ember application at runtime.
+
+## Goals
+
+- Define an interface for specifying where to load an asset needed for an application at runtime.
+- Allow asset URIs to be fingerprinted for caching reasons.
+- Support multiple inclusion of the same asset without duplicating the load.
+- Allow compositional loading of assets for future application segmentation.
+
+## Secondary / Related Goals
+
+- Keep the API of the forthcoming Asset Loader service simple.
+- Enable the prevention of duplicate requests for assets.
+
+# Detailed Design
+
+The default Asset Manifest generated by Ember-CLI will be a new file `asset-manifest.json`. Creating a standalone file, as opposed to an inline config or sub-section of another file, allows easy manipulation via additional build plugins (such as `broccoli-asset-rev`) and gives flexibility in how the contents are used. This file will by default, though optionally, be inserted into an application's `index.html` similarly to how `config/environment` is done today.
+
+At the core of the manifest is the idea of a "bundle" which is simply a collection of assets intended to be loaded together. Each bundle is associated with a "name" that represents the field it is under in the `bundles` property of the manifest. Even though `bundles` will be the only top-level field of the manifest in the first iteration, having the additional object layer will allow us to expand the manifest to include additional information in the future.
+
+Since we can access the package name of an Engine at runtime, regardless of aliasing, we should always load an Engine's assets according to its package name and not its mounted path. This will make it easier to avoid duplicate loading of assets and ensures we can support multiple inclusion of the same engine. Additionally, referencing bundles via a name rather than route path decouples asset loading from the routing behavior.
+
+Each bundle will have two primary fields: `assets` and `dependencies`.
+
+`assets` is an array of asset objects, which specify a `uri` and `type` for an asset to be loaded. The asset objects are sorted according to load order. This is important for assets such as `vendor.js` which need to be loaded before other assets like `app.js`.
```

Comment:
I added some additional language about ordering and the relationship between the manifest ordering and behavior of the loader.